easybuild/easyconfigs/i/IPython/IPython-5.8.0-fosscuda-2018b-Python-2.7.15.eb

```diff
@@ -0,0 +1,227 @@
+        'source_urls': ['https://pypi.python.org/packages/source/j/jupyter_core/'],
+        'use_pip': True,
+        'checksums': ['ba70754aa680300306c699790128f6fbd8c306ee5927976cbe48adacf240c0b7'],
+    }),
+    ('jupyter_client', '5.2.3', {
+        'source_urls': ['https://pypi.python.org/packages/source/j/jupyter_client/'],
+        'use_pip': True,
+        'checksums': ['27befcf0446b01e29853014d6a902dd101ad7d7f94e2252b1adca17c3466b761'],
+    }),
+    ('mistune', '0.8.4', {
+        'source_urls': ['https://pypi.python.org/packages/source/m/mistune/'],
+        'checksums': ['59a3429db53c50b5c6bcc8a07f8848cb00d7dc8bdb431a4ab41920d201d4756e'],
+    }),
+    ('terminado', '0.8.1', {
+        'source_urls': ['https://pypi.python.org/packages/source/t/terminado/'],
+        'checksums': ['55abf9ade563b8f9be1f34e4233c7b7bde726059947a593322e8a553cc4c067a'],
+    }),
+    ('futures', '3.2.0', {
+        'modulename': 'concurrent.futures',
+        'source_urls': ['https://pypi.python.org/packages/source/f/futures/'],
+        'checksums': ['9ec02aa7d674acb8618afb127e27fde7fc68994c0437ad759fa094a574adb265'],
+    }),
+    ('singledispatch', '3.4.0.3', {
+        'source_urls': ['https://pypi.python.org/packages/source/s/singledispatch/'],
+        'checksums': ['5b06af87df13818d14f08a028e42f566640aef80805c3b50c5056b086e3c2b9c'],
+    }),
+    ('backports_abc', '0.5', {
+        'source_urls': ['https://pypi.python.org/packages/source/b/backports_abc/'],
+        'checksums': ['033be54514a03e255df75c5aee8f9e672f663f93abb723444caec8fe43437bde'],
+    }),
+    ('tornado', '5.1.1', {
+        'source_urls': ['https://pypi.python.org/packages/source/t/tornado/'],
+        'checksums': ['4e5158d97583502a7e2739951553cbd88a72076f152b4b11b64b9a10c4c49409'],
+    }),
+    ('backcall', '0.1.0', {
+        'source_urls': ['https://pypi.python.org/packages/source/b/backcall/'],
+        'checksums': ['38ecd85be2c1e78f77fd91700c76e14667dc21e2713b63876c0eb901196e01e4'],
+    }),
+    ('prometheus_client', '0.4.2', {
+        'source_urls': ['https://pypi.python.org/packages/source/p/prometheus_client/'],
+        'checksums': ['046cb4fffe75e55ff0e6dfd18e2ea16e54d86cc330f369bebcc683475c8b68a9'],
+    }),
+    ('ipykernel', '4.10.0', {
+        'source_urls': ['https://pypi.python.org/packages/source/i/ipykernel/'],
+        'checksums': ['699103c8e64886e3ec7053f2a6aa83bb90426063526f63a818732ff385202bad'],
+    }),
+    ('notebook', '5.7.0', {
+        'source_urls': ['https://pypi.python.org/packages/source/n/notebook/'],
+        'use_pip': True,
+        'checksums': ['b85e4de3d54cf4f14fe1d0515a980ccb49ddd4cdd21250cc0d4fb6374d50b1a7'],
+    }),
+]
+
+modextrapaths = {'PYTHONPATH': ['lib/python%(pyshortver)s/site-packages']}
+
+sanity_check_paths = {
+    'files': ['bin/ipython'],
+    'dirs': ['lib/python%(pyshortver)s/site-packages/IPython'],
+}
+
+sanity_check_commands = [
+    "ipython -h",
+    "ipython notebook --help",
+    "iptest",
```

Comment:
after adding `vcversioner`, the sanity check is failing for me with

```
== FAILED: Installation ended unsuccessfully (build directory: /dev/shm/IPython/5.8.0/fosscuda-2018b-Python-2.7.15): build failed (first 300 chars): Sanity check failed: sanity check command iptest exited with code 1 (output: 
Test group: core
..............................................SSSSS....................S.S...................................................................................................................................
== Results of the build can be found in the log file(s) /hpctmp/ebapps/tmp/eb-bt5k38/easybuild-IPython-5.8.0-20190222.091002.JTPfG.log
ERROR: Build of /app1/ebapps/eb_files/gpu/cuda9/Python/IPython-5.8.0-fosscuda-2018b-Python-2.7.15.eb failed (err: 'build failed (first 300 chars): Sanity check failed: sanity check command iptest exited with code 1 (output: \nTest group: core\n..............................................SSSSS....................S.S...................................................................................................................................')
```

the log says

```
/app1/ebapps/easybuild/software/MPI/GCC-CUDA/7.3.0-2.30-9.2.88/OpenMPI/3.1.1/Python/2.7.15/bin/python: error while loading shared libraries: libcudart.so.9.2: cannot open shared object file: No such file or directory
```

but when I manually load `fosscuda` and ldd python it is found, so something is changing $LD_LIBRARY_PATH during the test?

I also see

```
UsageError: Line magic function `%doesntexist` not found.
.UsageError: Cell magic `%%doesntexist` not found.
................................................................................................S.........................................S...........................................................
======================================================================
FAIL: test_exit_code_signal_csh (IPython.core.tests.test_interactiveshell.TestSystemRaw)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/app1/ebapps/easybuild/software/MPI/GCC-CUDA/7.3.0-2.30-9.2.88/OpenMPI/3.1.1/IPython/5.8.0-Python-2.7.15/lib/python2.7/site-packages/IPython/core/tests/test_interactiveshell.py", line 568, in test_exit_code_signal_csh
    self.test_exit_code_signal()
  File "/app1/ebapps/easybuild/software/MPI/GCC-CUDA/7.3.0-2.30-9.2.88/OpenMPI/3.1.1/IPython/5.8.0-Python-2.7.15/lib/python2.7/site-packages/IPython/testing/decorators.py", line 213, in skipper_func
    return f(*args, **kwargs)
  File "/app1/ebapps/easybuild/software/MPI/GCC-CUDA/7.3.0-2.30-9.2.88/OpenMPI/3.1.1/IPython/5.8.0-Python-2.7.15/lib/python2.7/site-packages/IPython/core/tests/test_interactiveshell.py", line 561, in test_exit_code_signal
    self.assertEqual(ip.user_ns['_exit_code'], -signal.SIGALRM)
AssertionError: 127 != -14
    '127 != -14' = '%s != %s' % (safe_repr(127), safe_repr(-14))
    '127 != -14' = self._formatMessage('127 != -14', '127 != -14')
>>  raise self.failureException('127 != -14')
```

but I have no idea what's going on here...

Reply:
on a node with internet access (but other things are different, arch, OS version, etc.), the sanity check doesn't fail, weird...

the one that fails is on a CentOS 7.5.1804 Skylake system (with Volta GPUs, but this doesn't seem to be gpu related...).

Reply:
https://gist.github.com/migueldiascosta/bac25e88673926bb3af06974250cb3b9

Reply:
No problem here on CentOS 7 Skylake (but system does have internet access), see test report(s)

Reply:
@migueldiascosta Was `libcudart.so.9.2` available somewhere?

Reply:
@boegel yes, and

```bash
$ module load fosscuda/2018b
$ module load Python/2.7.15
$ ldd $EBROOTPYTHON/bin/python | grep cudart
	libcudart.so.9.2 => /app1/ebapps/easybuild/software/Compiler/GCC/7.3.0-2.30/CUDA/9.2.88/lib64/libcudart.so.9.2 (0x00007fb1a4a53000)
```

I don't know what's happening inside the sanity test... in particular, this only happens for `iptest`, if  I comment it everything is fine...

Reply:
@migueldiascosta Can you run a new test report on the system that had problems?

Reply:
@akesandgren sure, https://gist.github.com/migueldiascosta/09c2ada15e6fbcc072d90b1d29bc3898